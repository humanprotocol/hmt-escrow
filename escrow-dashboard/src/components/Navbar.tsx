import * as React from 'react';
import {
  Box,
  Link,
  Container,
  Menu,
  Toolbar,
  MenuItem,
  Button,
} from '@mui/material';
import { NavLink } from 'react-router-dom';
import { routes } from 'src/routes';

const Navbar: React.FC = (): React.ReactElement => {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);
  const handleOpenNavMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };
  const handleCloseNavMenu = (): void => {
    setAnchorEl(null);
  };

  return (
    <Box
      sx={{
        backgroundColor: 'primary.dark',
      }}
    >
      <Container maxWidth="xl">
        <Toolbar disableGutters>
          <Box sx={{ display: { xs: 'flex' } }}>
            <Button
              id="demo-positioned-button"
              aria-controls={open ? 'demo-positioned-menu' : undefined}
              aria-haspopup="true"
              aria-expanded={open ? 'true' : undefined}
              onClick={handleOpenNavMenu}
            >
              Jobs
            </Button>
            <Menu
              id="demo-positioned-menu"
              aria-labelledby="demo-positioned-button"
              anchorEl={anchorEl}
              open={open}
              onClose={handleCloseNavMenu}
              anchorOrigin={{
                vertical: 'top',
                horizontal: 'left',
              }}
              transformOrigin={{
                vertical: 'top',
                horizontal: 'left',
              }}
            >
              {routes.map((page) => (
                <MenuItem
                  key={page.key}
                  color="info.main"
                  onClick={handleCloseNavMenu}
                >
                  <Link
                    component={NavLink}
                    to={page.path}
                    color="primary.main"
                    underline="none"
                    variant="button"
                    sx={{ fontSize: 'large' }}
                  >
                    {page.title}
                  </Link>
                </MenuItem>
              ))}
            </Menu>
          </Box>
        </Toolbar>
      </Container>
    </Box>
  );
};

export default Navbar;
