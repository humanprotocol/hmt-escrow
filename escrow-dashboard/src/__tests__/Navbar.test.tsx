import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

describe('when rendered Navbar component', () => {
  it('should paste it into the greetings text', () => {
    render(<Navbar />);
  });
});

it('Navbar component renders correctly, corresponds to the snapshot', () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
