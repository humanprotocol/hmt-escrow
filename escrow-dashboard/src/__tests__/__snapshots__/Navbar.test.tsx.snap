// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiBox-root css-uolrcl"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
    >
      <div
        className="MuiBox-root css-hfc4ec"
      >
        <button
          aria-haspopup="true"
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          id="demo-positioned-button"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Jobs
        </button>
      </div>
    </div>
  </div>
</div>
`;
